elastic/kibana
# [ResponseOps] Mark cloud SMTP rate limiting and recipient address rejected errors as user errors

Resolves https://github.com/elastic/response-ops-team/issues/212

## Summary

There were some serverless alerts that were triggered by projects encountering rate limiting (SMTP 450) failures and failures due to the recipient address being rejected (SMTP 554) when using the Cloud SMTP preconfigured connector. Those errors are currently flagged as framework errors by default and this PR adds code to mark these as user errors.


### Checklist


### To verify

- You can use the cloud deploy from this PR and the following request to test the rate limiting error (SMTP 450) and verify the `errorSource: 'user'`. The cloud email service rate limits to 500 emails per 15 minutes.
```
curl -X POST -u $USERNAME:$PASSWORD $CLOUD_URL/api/actions/connector/elastic-cloud-email/_execute \
    -H 'kbn-xsrf: true' \
    -H 'Content-Type: application/json' \
    -d '
      { 
        "params": {
          "to": ["$YOUR_EMAIL"],
          "subject": "testing email",
          "message": "test"
        }
      }
    '
```

- I haven't been able to test the SMTP 554, but it should work the same as the SMTP 450 error.
